third_party/pcsc-lite/naclport/server_clients_management/src/client_request_processor.cc

```diff
@@ -400,6 +405,28 @@ GenericRequestResult PcscLiteClientRequestProcessor::SCardConnect(
     const std::string& reader_name,
     DWORD share_mode,
     DWORD preferred_protocols) {
+  SCARDHANDLE s_card_handle;
+  DWORD active_protocol;
+  LONG return_code = SCardConnect(
+      s_card_context, reader_name, share_mode, preferred_protocols,
+      &s_card_handle, &active_protocol, /*update_card_handles_registry=*/true,
+      /*handle_protocol_mismatch_error=*/true);
+
+  if (return_code != SCARD_S_SUCCESS)
+    return ReturnValues(return_code);
+
+  return ReturnValues(return_code, s_card_handle, active_protocol);
+}
+
+LONG PcscLiteClientRequestProcessor::SCardConnect(
+    SCARDCONTEXT s_card_context,
+    const std::string& reader_name,
+    DWORD share_mode,
+    DWORD preferred_protocols,
+    LPSCARDHANDLE s_card_handle,
+    LPDWORD active_protocol,
+    bool update_card_handles_registry,
+    bool handle_protocol_mismatch_error) {
   FunctionCallTracer tracer("SCardConnect", logging_prefix_,
```

Comment:
Not sure if it's the best place for this logging. Currently we log all calls exactly as they come, without any modifications. If we start logging "SCardConnect" in cases when the caller didn't actually call it, it'll make the investigation harder.

How about moving this `FunctionCallTracer` back to the original (public) `SCardConnect()` method?